###
#
# @host: Primary_Core_Zabbix
# 
# @author: Sai
# @date:  10-Oct-2020
# @version: V.1
#
###
  
  - setup:
  - set_fact: 
      Admin_Root: "{{lookup('env','Admin_Root')}}"
#core server ip
  - name: Print the name of core node
    set_fact:
      Zabbix_Server: "{{ item }}"
    with_items:
      - "{{ groups ['Secondary_Core_Zabbix'] }}"

#smtp
  - name: Running script for onboard
    command: python3.7 Zabbix_create_users.py "Secondary" "{{ Zabbix_Server }}"
    args:
      chdir: "{{ ansible_env.Admin_Root }}/zabbix/Bin/Configurations"

# Added this for rollback ---------------------------------
    register: result
    ignore_errors: yes

  - set_fact:
      pyresult: "3rr0rC0d3[00{{result.rc}}]"

  - debug:
      msg: "{{result.stdout_lines[0]}}"

  - set_fact:
      pyresult: "{{result.stdout_lines[0]}}"
    when: result.rc == 0


